submission/programs/adae.R

```diff
@@ -0,0 +1,350 @@
+  # (USUBJID), System Organ Class (AEBODSYS), Start Date                #
+  # (ASTDT), and Sequence Number (AESEQ) and flag the                   #
+  # first record (set AOCC03FL=’Y’) within each Subject and             #
+  # SOC                                                                 #
+  #---------------------------------------------------------------------#
+  restrict_derivation(
+    derivation = derive_var_extreme_flag,
+    args = params(
+      by_vars = vars(USUBJID, AEBODSYS   ),
+      order = vars(  ASTDT, AESEQ),
+      new_var = AOCC03FL ,
+      mode = "first"
+    ), filter = TRTEMFL == "Y" & AESER =='Y'
+  )  %>% 
+
+  #---------------------------------------------------------------------#
+  # AOCC04FL - 1st Occurrence 04 Flag for Serious PT                    #
+  #---------------------------------------------------------------------#
+  # Subset to TRTEMFL='Y' and AESER='Y' and sort by Subject             #
+  # (USUBJID), System Organ Class (AEBODSYS), Preferred                 #
+  # Term (AEDECOD), Start Date (ASTDT), and Sequence                    #
+  # Number (AESEQ) and flag the first record (set                       #
+  # AOCC04FL=’Y’) within each Subject, SOC, and PT                      #
+  #---------------------------------------------------------------------#  
+  restrict_derivation(
+    derivation = derive_var_extreme_flag,
+    args = params(
+      by_vars = vars(USUBJID, AEBODSYS , AEDECOD  ),
+      order = vars(  ASTDT, AESEQ),
+      new_var = AOCC04FL ,
+      mode = "first"
+    ), filter = TRTEMFL == "Y" & AESER =='Y'
+  )  %>% 
+
+  # --------------------------------------------------------------------#
+  # CQ01NAM - Customized Query 01 Name                                  #
+  # --------------------------------------------------------------------#
+  # If AEDECOD contains any of the character strings of                 #
+  # ('APPLICATION', 'DERMATITIS', 'ERYTHEMA', 'BLISTER')                #
+  # OR if AEBODSYS='SKIN AND SUBC UTANEOUS TISSUE                       #
+  # DISORDERS' but AEDECOD is not in ('COLD SWEAT',                     #
+  #                                   'HYPERHIDROSIS', 'ALOPECIA') then #
+  # CQ01NAM='DERMATOLOGIC EVENTS' Otherwise  CQ01NAM=NULL               #
+  # --------------------------------------------------------------------#  
+  mutate ( CQ01NAM  = ifelse(str_detect(.$AEDECOD , "APPLICATION" ) |
+                             str_detect(.$AEDECOD , "DERMATITIS")   |
+                             str_detect(.$AEDECOD , "ERYTHEMA")     |
+                             str_detect(.$AEDECOD , "BLISTER")      | 
+                             str_detect(.$AEBODSYS , "SKIN AND SUBCUTANEOUS TISSUE DISORDERS") &
+                             !str_detect(.$AEDECOD , "COLD SWEAT|HYPERHIDROSIS|ALOPECIA")  , 
+                             "DERMATOLOGIC EVENTS" , 
+                             NA_character_) 
```

Comment:
The `.$` prefix is not required here.

Reply:
removed.

Reply:
Please review and resolve to ensure you do not see it on your end.

Reply:
@SHAESEN2 or @thomas-neitmann If ok, please `Resolve conversation`.